build-tools/docker/example-mainnet/.env

```diff
@@ -0,0 +1,36 @@
+# This will be used as a prefix for container and volume names
+# (if not specified, `docker compose` will use the name of the parent directory as the project name).
+COMPOSE_PROJECT_NAME=mintlayer-mainnet
+
+# Dockerhub username, from which the docker images will be pulled.
+DOCKERHUB_USERNAME=mintlayer
+# The version of mintlayer binaries to use.
+# This will be used together with DOCKERHUB_USERNAME to form full names of docker images
+# (i.e. $DOCKERHUB_USERNAME/image_name:$ML_SOFTWARE_VERSION)
+ML_SOFTWARE_VERSION=0.3.0
+
+# User name and db name for the postgres dbms use by the api server. Since the entire dbms will be
+# run in a dedicated docker container using a dedicated docker volume, the names don't matter much.
+# Just note that the db will be created when the volume is first initialized, so changing
+# the name later won't work.
+API_SERVER_POSTGRES_USER=postgres
+API_SERVER_POSTGRES_DB=postgres
+# The password for the postgres dbms.
+API_SERVER_POSTGRES_PASSWORD=use-strong-password
+
+# The node's rpc and p2p ports will be mapped to these ports on the host machine.
+NODE_RPC_HOST_PORT=3030
+NODE_P2P_HOST_PORT=3031
```

Comment:
It's a bad idea to provide that option. People will think they can change this. They shouldn't.

Now I understand that when a proxy is used, one should set this at the external network interface, but if we want to keep this parameter accessible, we should put a big warning above it that the external port at the routable ip address (in case it's desired to reach this server from the outside) must be 3031, otherwise it won't be reachable.

Reply:
Actually, this should work if the listening port inside the container is the same as the one exposed on the host machine. I've made the corresponding changes in docker-compose.yml and updated the comment there.

Reply:
Like explained before, this depends on whether you're doing network=host, and if the container behind its own NAT (which is the case if network=host isn't mentioned), changing the port won't have an effect. There's more to this than simply changing the port.

Reply:
[discussed] I've added a warning comment to `.env`  explaining that changing this port value may not be enough.